libp2p/handler_test.go

```diff
@@ -0,0 +1,68 @@
+package libp2p
+
+import (
+	"context"
+	"math"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	libp2p "github.com/libp2p/go-libp2p"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	net "github.com/libp2p/go-libp2p-net"
+	"github.com/status-im/whisper/whisperv6"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	proto = "/shh/6.0"
+)
+
+func TestWhisperOverLibp2p(t *testing.T) {
```

Comment:
I would love to see this test a bit more split into sections and with more detailed variable names. I feel like this is the most important one to understand how this works.

Here is what I mean:
```go
// setting up libp2p node with Whisper
node1, err := libp2p.New(context.Background(), ...)
w := whisperv6.New(&whisperv6.Config{ ... })
node1.SetStreamHandler(proto, func(s net.Stream) { ... }

// create another libp2p node1 as a peer
node2, err := libp2p.New(context.Background(), ...)
node2.Peerstore().AddAddr(h.ID(), h.Addrs()[1], 30*time.Second)
s, err := node2.NewStream(context.TODO(), h.ID(), proto)
msgs := NewStream(s, 2*time.Second, 0)

// and so on
```